### DISCUSSION TITLE
Only number accepted for DataGridTextColumn

### DISCUSSION THREAD

**Discussion by hxhgxy** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3158] [Ideas]
I made attach property for TextBox to accept numbers only. How can I apply this to DataGridTextColumn?
Thanks,

`              public static bool GetIsOnlyNumber(DependencyObject obj)
		{
			return (bool)obj.GetValue(IsOnlyNumberProperty);
		}

		public static void SetIsOnlyNumber(DependencyObject obj, bool value)
		{
			obj.SetValue(IsOnlyNumberProperty, value);
		}

		public static readonly DependencyProperty IsOnlyNumberProperty =
		    DependencyProperty.RegisterAttached("IsOnlyNumber", typeof(bool), typeof(TextBox), new PropertyMetadata(false,
		        (s, e) =>
		        {
			        if (s is TextBox textBox)
			        {
				        textBox.SetValue(InputMethod.IsInputMethodEnabledProperty, !(bool)e.NewValue);
				        textBox.PreviewTextInput -= TxtInput;
				        if ((bool)e.NewValue)
				        {
					        textBox.PreviewTextInput += TxtInput;
				        }
			        }
		        }));

		private static void TxtInput(object sender, System.Windows.Input.TextCompositionEventArgs e)
		{
			e.Handled = new Regex("[^0-9.]+").IsMatch(e.Text);
		}
`

**Comment by nicolaihenriksen** [comment]
@hxhgxy You should be able to achieve this by applying your attached property to the style used for the "editing element". When a data grid cell of a `DataGridTextColumn` enters edit mode, it displays a `TextBox` and that `TextBox` is then styled using the `DataGridTextColumn.EditingElementStyle`.

So you should define a named `Style` targeting a `Textbox` and then apply your attached property in a `<Setter />` on that style. You then need to point to that style in the `EditingElementStyle` of your `DataGridTextColumn`.

**Comment by hxhgxy** [comment]
Thanks. If so, it is easier to use DataGridTemplateColumn, right? I'm using TextBox under DataGridTemplateColumn and set the attached property on TextBox currently.

**Reply by nicolaihenriksen** [reply]
If you find that easier, then of course you can do that 😃 There are typically multiple ways of achieving the same goal.

Note: you could possibly be "missing out" on some built-in features of the `DataGrid` when using a `DataGridTemplateColumn`, because it now lacks knowledge of the contents in the column. Here I am thinking sorting, and stuff like that. I don't know for sure though, so you would need to experiment with that on your own.
